Makefile

```diff
@@ -226,6 +226,7 @@ docker-image-no-clean: GIT_VERSION
 		--build-arg LOCKDEBUG=${LOCKDEBUG} \
 		--build-arg V=${V} \
 		--build-arg LIBNETWORK_PLUGIN=${LIBNETWORK_PLUGIN} \
+		--build-arg CILIUM_SHA=$$(cat GIT_VERSION | cut -d" " -f1 | tr -d '\n') \
```

Comment:
Could this be set with `docker build --label` in stead?

Reply:
I think you what you are looking for is `$(GIT_VERSION)`, it should be set already.
I'd just use the whole string, i.e. `--label "cilium_git_version=$(GIT_VERSION)"`, or otherwise `--label "cilium_git_version=$(firstword $(GIT_VERSION))"` if you prefer to use just the short commit hash.

Reply:
We cannot use `--label` here, because it only labels the final image. We want to label intermediate images too so we know which ones to prune.

Good catch with `firstword` usage, I didn't know I can use it like that, will change.

Reply:
I see, seems suboptimal but acceptable :)